service/src/main/scala/za/co/absa/loginsvc/rest/config/provider/ConfigProvider.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.loginsvc.rest.config.provider
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import pureconfig._
+import pureconfig.generic.auto._
+import pureconfig.module.yaml._
+import za.co.absa.loginsvc.rest.config.actuator.GitConfig
+import za.co.absa.loginsvc.rest.config.auth._
+import za.co.absa.loginsvc.rest.config.validation.ConfigValidationException
+import za.co.absa.loginsvc.rest.config.{BaseConfig, JwtConfig}
+
+import java.nio.file.{InvalidPathException, Paths}
+
+@Component
+class ConfigProvider(@Value("${DefaultYamlPath:service/src/main/resources/application.yaml}") yamlContent: String)
```

Comment:
Also, how do you handle config overrides?
In the previous version, one could pass `spring.config.location=/path/to/config`. Here, it seems you hardcode a path to `application.yaml` that is packaged inside the war/jar.

We need to be able to build the package without an internal config and pass an external one somehow. Sorry if you have thought about this and I am missing it.

https://pureconfig.github.io/docs/ talks about using `ConfigSource.default` as a way to define an overridable source, then e.g. `-Dconfig.file=path/to/config-file` can be used for externilized config.

... in the end, if Yaml was too hard for us, with pureConfig/tsConfig, we could be using HOCON. That's not a problem, since it also allows using objects in config (properties do not). _This would be much cheaper to do now than later._

Reply:
One more point here: later, in when dockerized, we need to slip the configuration to the running war, too. This we currently do via setting env var `SPRING_CONFIG_ADDITIONAL-LOCATION=/opt/application.yaml`

Bear this in mind that some form of configuration override is probably needed as well. Does pureConfig support Typesafe config's `CONFIG_FILE` env property for application.conf external loading?

Reply:
Ok, so @Value in Spring checks for values matching that key in config files, environment variables and program arguments.

So I've changed it so that it mimics spring.config.location which can be set in either program arguments or environment variables, so this should sort that out. I've added this to the readme.